h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -539,56 +568,41 @@ private void addGridKey(Key<Grid> gridKey) {
    * @param baseParms ModelBuilder parameter values that are common across all models in the search
    * @param searchParms hyperparameter search space
    * @return the started hyperparameter search job
    */
-  public Job<Grid> hyperparameterSearch(String algoName, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
-    return hyperparameterSearch(null, algoName, baseParms, searchParms);
+  public Job<Grid> hyperparameterSearch(Algo algo, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
+    return hyperparameterSearch(null, algo, baseParms, searchParms);
   }
 
   /**
    * Do a random hyperparameter search.  Caller must eventually do a <i>get()</i>
    * on the returned Job to ensure that it's complete.
    * @param gridKey optional grid key
-   * @param algoName name of the algo, e.g. "GBM"; used for messages and for building the grid key if it's not specified
+   * @param algo  the Algo, e.g. "GBM"; used for messages and for building the grid key if it's not specified
    * @param baseParms ModelBuilder parameter values that are common across all models in the search
    * @param searchParms hyperparameter search space
    * @return the started hyperparameter search job
    */
-  public Job<Grid> hyperparameterSearch(Key<Grid> gridKey, String algoName, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
-    if (ArrayUtils.contains(skipAlgosList, algoName)) {
-      userFeedback.info(Stage.ModelTraining,"AutoML: skipping algo " + algoName + " hyperparameter search");
-      return null;
-    }
+  public Job<Grid> hyperparameterSearch(Key<Grid> gridKey, Algo algo, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
+    if (exceededSearchLimits(algo, JobType.HyperparamSearch)) return null;
 
     setCommonModelBuilderParams(baseParms);
 
-    if (remainingModels() <= 0) {
-      userFeedback.info(Stage.ModelTraining,"AutoML: hit the max_models limit; skipping " + algoName + " hyperparameter search");
-      return null;
-    }
-
     HyperSpaceSearchCriteria.RandomDiscreteValueSearchCriteria searchCriteria = (HyperSpaceSearchCriteria.RandomDiscreteValueSearchCriteria)buildSpec.build_control.stopping_criteria.clone();
     if (searchCriteria.max_runtime_secs() == 0)
-      searchCriteria.set_max_runtime_secs(this.timeRemainingMs() / 1000.0);
+      searchCriteria.set_max_runtime_secs(Math.round(timeRemainingMs() / 1000.0));
     else
-      searchCriteria.set_max_runtime_secs(Math.min(searchCriteria.max_runtime_secs(),
-              timeRemainingMs() / 1000.0));
-
-    if (searchCriteria.max_runtime_secs() <= 0.001) {
-      userFeedback.info(Stage.ModelTraining,"AutoML: out of time; skipping " + algoName + " hyperparameter search");
-      return null;
-    }
+      searchCriteria.set_max_runtime_secs(Math.min(searchCriteria.max_runtime_secs(), Math.round(timeRemainingMs() / 1000.0)));
```

Comment:
Same as above. Any reason for removing `UserFeedback`?

Reply:
closed